server/market/market.go

```diff
@@ -1007,7 +1009,8 @@ func (m *Market) processReadyEpoch(ctx context.Context, epoch *readyEpoch) {
 
 	// Perform order matching using the preimages to shuffle the queue.
 	m.bookMtx.Lock()
-	matchTime := time.Now() // considered as the time at which matched cancel orders are executed
+	m.swapper.BeginMatchAndNegotiate() // Suspend Swapper's order completion checking.
+	matchTime := time.Now()            // considered as the time at which matched cancel orders are executed
 	seed, matches, _, failed, doneOK, partial, booked, unbooked, updates := m.matcher.Match(m.book, ordersRevealed)
```

Comment:
I'm not particularly happy about this solution, which is required because order filled/remaining amounts are updated here in market before the Swapper records the match.  Another solution, which doesn't sound great either since it's full of issues, is to add a field to each order to account for the amount successfully swapped.

Reply:
Regarding the linked comment.

> We may even wish to incentivize the user to complete remaining swaps

Because operators will ultimately have control over undoing penalties, the user is already incentivized to complete other swaps, because operators should be able to examine the client's trading history when considering reversion, but...

`AuthManager` [boots clients](https://github.com/decred/dcrdex/blob/7565aa09f25e948c9e54cc519632488b7ed023dc/server/auth/auth.go#L266) when they are penalized. They wouldn't be allowed to reconnect with a closed account either. 

A proper solution will probably involve some wider policy changes and a major spec update.